app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryTest.kt

```diff
@@ -8,34 +8,49 @@ import com.lightricks.feedexercise.database.UserProject
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
 import org.junit.Assert
-import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
-class FeedRepositoryTest(private var db : FeedDatabase,
-                         private var feedRepository: FeedRepository,
-                         private var userProjectList : List<UserProject>) {
-
-    @Before
-    fun init(){
-        this.db = Room.inMemoryDatabaseBuilder(ApplicationProvider.getApplicationContext(),
-            FeedDatabase::class.java).build()
-        this.feedRepository = FeedRepository(MockApiService,db)
+class FeedRepositoryTest {
+    private val db: FeedDatabase = Room.inMemoryDatabaseBuilder(
+        ApplicationProvider.getApplicationContext(),
+        FeedDatabase::class.java
+    ).build()
+    private val feedRepository: FeedRepository = FeedRepository(ConstantItemsApiService, db)
+    private val userProjectList: List<UserProject> = Constans.hardCodedList.map { item ->
+        UserProject(item.id, responseUrlToThumbnailUrl(item.thumbnailUrl), item.isPremium)
     }
 
+
     @Test
-    fun testRefreshSaves(){
+    fun testRefreshSaves() {
+        var fetchedList: List<UserProject>
+        feedRepository.getAllProjects().doOnNext { userProjects ->
+            fetchedList = userProjects
+            Assert.assertTrue(fetchedList.isEmpty())
+        }.subscribe()
         feedRepository.refresh().test().awaitTerminalEvent()
-        var fetchedList : List<UserProject>
         db.userProjectDao().getAll().subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
-            .doOnNext{userProjects ->
+            .doOnNext { userProjects ->
                 fetchedList = userProjects
-                Assert.assertEquals(fetchedList,userProjectList)
+                Assert.assertEquals(fetchedList, userProjectList)
             }
             .subscribe()
     }
+
+    @Test
+    fun testGetAll() {
+        this.db.userProjectDao().insertAll(userProjectList).test().awaitTerminalEvent()
+        var fetchedList: List<UserProject>
+        feedRepository.getAllProjects().subscribe { fetchedUserProjects ->
+            fetchedList = fetchedUserProjects
+            Assert.assertEquals(fetchedList, userProjectList)
+        }
```

Comment:
Also, I'm afraid the test can finish before `fetchedUserProjects` are fetched.
You should call `test().awaitTerminalEvent()`.